scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -297,21 +334,27 @@ async fn update_list<'a>(
 
     // We use the same query for updating and deleting values.
     // In case of deleting, we simply set the value to null.
-    let update_query = Query::new(format!(
-        "UPDATE {ks}.{tbl} USING TTL ? SET {list}[SCYLLA_TIMEUUID_LIST_INDEX(?)] = ? WHERE {cond}",
-        ks = destination_table_params.dest_keyspace_name,
-        tbl = destination_table_params.dest_table_name,
-        list = column_name,
-        cond = destination_table_params.keys_cond
-    ));
+    let update_query = precomputed_queries.non_key_column_queries
+        .entry((column_name.to_string(), QueryType::UpdateListElements))
+        .or_insert(
+            destination_table_params.dest_session.prepare(
+                format!(
+                    "UPDATE {ks}.{tbl} USING TTL ? SET {list}[SCYLLA_TIMEUUID_LIST_INDEX(?)] = ? WHERE {cond}",
+                    ks = destination_table_params.dest_keyspace_name,
+                    tbl = destination_table_params.dest_table_name,
+                    list = column_name,
+                    cond = destination_table_params.keys_cond
+                )
+            ).await?
+        );
```

Comment:
Moreover, I think that it would make sense to convert the `PrecomputedQueries` struct from being just a data container and promote it into some kind of a cache of prepared statements used by the replicator. It would be responsible for preparing, looking up and executing queries with given parameters, along with formatting the strings (just like the code highlighted above).

It could probably keep the `DestinationTableParams` inside itself, I don't it used outside of that context so IMO it should be fine. By doing this you could also probably avoid converting so many methods into standalone functions and reduce their number of parameters.